recirq/qcqmc/afqmc_circuits.py

```diff
@@ -0,0 +1,413 @@
+"""Trial wavefunction circuit ansatz primitives."""
+
+import itertools
+from typing import Iterable, List, Tuple
+
+import cirq
+import numpy as np
+import openfermion
+from openfermion.circuits.primitives.state_preparation import (
+    _ops_from_givens_rotations_circuit_description,
+)
+from openfermion.linalg.givens_rotations import givens_decomposition
+from openfermion.linalg.sparse_tools import jw_sparse_givens_rotation
+
+
+class GeminalStatePreparationGate(cirq.Gate):
+    def __init__(self, angle: float, indicator: bool = False):
+        """A 4-qubit gate that takes to produce a geminal state
+
+        This state takes |00b0〉to sin(θ) |0011〉+ cos(θ) |1100〉and |00(1-b)0〉 to |0000〉.
+
+        The perfect pairing operator T_pp (eqn C5) is a sum over pairs. Each
+        pair consists of four qubits (two spin states; occupied and unoccupied
+        for each), and is a rotation of the form implemented by this gate.
+
+        Given qubits a, b, c, d: This composite gate decomposes into a
+        Givens rotation on qubits a, c followed by two CNOTs copying a to b
+        and c to d.
+
+        Args:
+            angle: The angle θ.
+            indicator: The bit b. TODO(wjhuggins) how is this used in the experiment?
```

Comment:
@wjhuggins :D

but yeah: remove or open an issue and link it; probably from the body of the docstring rather than inline in the Args section

Reply:
Done.